src/terms/Hamiltonian.jl

```diff
@@ -203,14 +206,32 @@ end
         end
         ret
     end
+    scratch = _ham_allocate_scratch(basis)
     hks_per_k = [flatten([blocks[ik] for blocks in operators])
                  for ik = 1:length(basis.kpoints)]      # hks_per_k[ik][it]
-
-    ham = Hamiltonian(basis, [HamiltonianBlock(basis, kpt, hks)
+    ham = Hamiltonian(basis, [HamiltonianBlock(basis, kpt, hks; scratch)
                               for (hks, kpt) in zip(hks_per_k, basis.kpoints)])
     energies = Energies(term_names, energy_values)
     (; energies, ham)
 end
+
+"""
+Faster version than energy_hamiltonian for cases where only the energy is needed.
+"""
+@timing function energy(basis::PlaneWaveBasis, ψ, occupation; kwargs...)
```

Comment:
while you're at it define a fallback `energy(term, ...)` which calls ene_ops, and use it here